src/BuildOnWindows.cmake

```diff
@@ -53,31 +57,101 @@ else()
     # into torch_xpu_ops_sycl_unary_binary_kernels.dll.
     string(REGEX MATCH "Pow" IS_POW ${sycl_src})
     string(REGEX MATCH "Copy" IS_COPY ${sycl_src})
-    if(IS_BINARY STREQUAL "" AND IS_UNARY STREQUAL "" AND IS_POW STREQUAL "" AND IS_COPY STREQUAL "")
-      list(APPEND ATen_XPU_SYCL_NON_UNARY_BINARY_SRCS ${sycl_src})
+    string(REGEX MATCH "Activation" IS_ACTIVATION ${sycl_src})
+    string(REGEX MATCH "Foreach" IS_FOREACH ${sycl_src})
+    string(REGEX MATCH "Reduce" IS_REDUCE ${sycl_src})
+
+    if(NOT IS_FOREACH STREQUAL "")
+      list(APPEND ATen_XPU_SYCL_FOREACH_SRCS ${sycl_src})
+    elseif(NOT IS_BINARY STREQUAL "")
+      list(APPEND ATen_XPU_SYCL_BINARY_SRCS ${sycl_src})
+    elseif(NOT IS_UNARY STREQUAL "" OR NOT IS_COPY STREQUAL "" OR NOT IS_POW STREQUAL "")
+      list(APPEND ATen_XPU_SYCL_UNARY_SRCS ${sycl_src})
+    elseif(NOT IS_REDUCE STREQUAL "")
+      list(APPEND ATen_XPU_SYCL_REDUCE_SRCS ${sycl_src})
+    elseif(NOT IS_ACTIVATION STREQUAL "")
+      list(APPEND ATen_XPU_SYCL_ACTIVATION_SRCS ${sycl_src})
     else()
-      list(APPEND ATen_XPU_SYCL_UNARY_BINARY_SRCS ${sycl_src})
+      list(APPEND ATen_XPU_SYCL_OTHERS_SRCS ${sycl_src})
     endif()
   endforeach()
 
-  set(sycl_unary_binary_lib torch_xpu_ops_sycl_unary_binary_kernels)
+  # Binary kernel lib
+  set(sycl_binary_lib torch_xpu_ops_sycl_binary_kernels)
```

Comment:
You can try the interface library to make here explicit
https://cmake.org/cmake/help/latest/command/add_library.html#interface-libraries

Reply:
Could you elaborate the suggestion? Interface library won't involve building, but a cmake target to hold some properties?

Reply:
you can set an interface library like ```ATen_XPU_SYCL_SRCS``` and link all separated libraries into the interface library. then you only need to link the interface library to ```torch_xpu_ops_aten```
next time if you have more new libraries, you only need to link the new library to the interface, don't need to change the interface linking.

Reply:
here xetla_kernels is an interface library https://github.com/intel/intel-extension-for-pytorch/blob/8d4670e4c131a476c6aabb686f33688ca9949243/csrc/gpu/aten/operators/xetla/kernels/CMakeLists.txt#L117
where it is used https://github.com/intel/intel-extension-for-pytorch/blob/8d4670e4c131a476c6aabb686f33688ca9949243/csrc/gpu/CMakeLists.txt#L249

Reply:
Look good to me to make the logic simple. BTW, I need additional logic like not only,
`target_link_library(torch_xpu_ops_aten, torch_xpu_ops_kernel_itf)`, But also,
`target_link_library(torch_xpu_ops_kernel1, torch_xpu)`
`target_link_library(torch_xpu_ops_kernel2, torch_xpu)`
`target_link_library(torch_xpu_ops_kernel3, torch_xpu)`
So for part 1, I can use interface library, for part 2, I have to config link library separately.

Reply:
interface support bidirectional linking
https://github.com/intel/intel-extension-for-pytorch/blob/8d4670e4c131a476c6aabb686f33688ca9949243/csrc/gpu/CMakeLists.txt#L118

Reply:
LGTM. But so far the current impl looks good to me, as only the code piece uses these targets. If these targets need to be used in many places, interface target will be the choice. Thanks.